protocol/x/vault/keeper/shares.go

```diff
@@ -90,3 +92,104 @@ func (k Keeper) GetAllOwnerShares(ctx sdk.Context) []types.OwnerShare {
 	}
 	return allOwnerShares
 }
+
+// GetLockedShares gets locked shares for an owner.
+func (k Keeper) GetLockedShares(
+	ctx sdk.Context,
+	owner string,
+) (val types.LockedShares, exists bool) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), []byte(types.LockedSharesKeyPrefix))
+
+	b := store.Get([]byte(owner))
+	if b == nil {
+		return val, false
+	}
+
+	k.cdc.MustUnmarshal(b, &val)
+	return val, true
+}
+
+// SetLockedShares sets locked shares for an owner.
+func (k Keeper) SetLockedShares(
+	ctx sdk.Context,
+	owner string,
+	lockedShares types.LockedShares,
+) error {
+	if err := lockedShares.Validate(); err != nil {
+		return err
+	}
+
+	b := k.cdc.MustMarshal(&lockedShares)
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), []byte(types.LockedSharesKeyPrefix))
+	store.Set([]byte(owner), b)
+
+	return nil
+}
+
+// GetAllLockedShares gets all locked shares.
+func (k Keeper) GetAllLockedShares(ctx sdk.Context) []types.LockedShares {
+	allLockedShares := []types.LockedShares{}
+	lockedSharesStore := prefix.NewStore(ctx.KVStore(k.storeKey), []byte(types.LockedSharesKeyPrefix))
+	lockedSharesIterator := storetypes.KVStorePrefixIterator(lockedSharesStore, []byte{})
+	defer lockedSharesIterator.Close()
+	for ; lockedSharesIterator.Valid(); lockedSharesIterator.Next() {
+		var lockedShares types.LockedShares
+		k.cdc.MustUnmarshal(lockedSharesIterator.Value(), &lockedShares)
+		allLockedShares = append(allLockedShares, lockedShares)
+	}
+	return allLockedShares
+}
+
+// LockShares locks `sharesToLock` for `ownerAddress` until height `tilBlock`.
```

Comment:
nit: Should there be a TODO to compare the `sharesToLock` to the total number of shares owned by the `ownerAddress`?